src/system/MailerModule/Api/MailerApi.php

```diff
@@ -0,0 +1,271 @@
+     * @param PermissionApi            $permissionApi   PermissionApi service instance.
+     */
+    public function __construct(TranslatorInterface $translator, EventDispatcherInterface $eventDispatcher, array $mailerParams, VariableApi $variableApi, Swift_Mailer $mailer, PermissionApi $permissionApi)
+    {
+        $this->setTranslator($translator);
+        $this->eventDispatcher = $eventDispatcher;
+        $this->mailer = $mailer;
+        $this->permissionApi = $permissionApi;
+
+        $modVars = $variableApi->getAll('ZikulaMailerModule');
+        $this->dataValues = array_merge($mailerParams, $modVars);
+    }
+
+    /**
+     * Sets the translator.
+     *
+     * @param TranslatorInterface $translator Translator service instance.
+     */
+    public function setTranslator(/*TranslatorInterface */$translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * API function to send e-mail message.
+     * It is assumed that basic parameters for sender and recipient(s) have already been set.
+     *
+     * @param Swift_Message $message The message object.
+     * @param string        $subject message subject
+     * @param string        $body message body, if altbody is provided then
+     *                            this is the HTML version of the body
+     * @param string        $altBody alternative plain-text message body, if specified the
+     *                               e-mail will be sent as multipart/alternative
+     * @param bool          $html HTML flag, if altbody is not specified then this
+     *                            indicates whether body contains HTML or not; if altbody is
+     *                            specified, then this value is ignored, the body is assumed
+     *                            to be HTML, and the altbody is assumed to be plain text
+     * @param array         $headers custom headers to add
+     * @param array         $attachments array of either absolute filenames to attach
+     *                                   to the mail or array of arrays in format
+     *                                   array($path,$filename,$encoding,$type)
+     * @param array         $stringAttachments array of arrays to treat as attachments, format array($string,$filename,$encoding,$type)
+     * @param array         $embeddedImages array of absolute filenames to image files to embed in the mail
+     *
+     * @throws \RuntimeException Thrown if there's an error sending the e-mail message
+     *
+     * @return bool true if successful
+     */
+    public function sendMessage(Swift_Message $message, $subject, $body, $altBody, $html, array $headers = array(), array $attachments = array(), array $stringAttachments = array(), array $embeddedImages = array())
+    {
+        // Allow other bundles to control mailer behavior
+        $event = new GenericEvent($this, $args);
+        $this->eventDispatcher->dispatch('module.mailer.api.sendmessage', $event);
```

Comment:
will you please create a `MailerEvents` class and declare a `const` with the event name (see similar in new users module and elsewhere). This will help 'self document' events that are available.